### PR TITLE
TS-5006: Fix CID 1356975

### PR DESCRIPTION
**Pull request by PSUdaemon**
Check for NULL before pointer dereference.

### CODE REVIEW

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/1083/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/976/ for details.

**Comment by zwoop** [comment]
This needs a Jira number.

File: proxy/http/HttpSM.cc [hunk 1]
```diff
@@ -3309,7 +3309,7 @@ HttpSM::tunnel_handler_ua(int event, HttpTunnelConsumer *c)
 
     // only external POSTs should be subject to this logic; ruling out internal POSTs here
     bool is_eligible_post_request = (t_state.method == HTTP_WKSIDX_POST);
-    if (is_eligible_post_request) {
+    if (is_eligible_post_request && ua_session) {
```
Comment (zwoop) [inline proxy/http/HttpSM.cc]: I don't know this code much, but seeing that we don't crash here, presumably ua_session generally is not NULL here. So maybe a better thing is a release assert?

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/1093/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/985/ for details.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/1100/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/993/ for details.

**Pull request merged**